modules/backup_replication/variables.tf

```diff
@@ -0,0 +1,39 @@
+variable "resource_name_prefix" {
+  description = "Resource name prefix used for tagging and naming AWS resources"
+  type        = string
+
+  validation {
```

Comment:
This validation should be handled at the root module, not sure if it is needed here.